docs/articles/documentation/extending-testcafe/custom-reporter/reporter-methods.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: docs
+title: Reporter Methods
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/reporter-methods.html
+checked: false
+---
+# Reporter Methods
+
+To create your [custom reporter](index.md#implementing-the-reporter), you need to implement the following reporter methods.
+
+> You can use [helper methods](helper-methods.md) within the reporter methods to output required data.
+
+## reportTaskStart
+
+Fires when a test task starts, which happens at the beginning of testing.
+
+```text
+reportTaskStart (startTime, userAgents, testCount)
+```
+
+Parameter    | Type               | Description
+------------ | ------------------ | ---------------------------------------
+`startTime`  | Date/Time          | The date and time when testing started.
+`userAgents` | An array of string | The list of browsers used for testing.
+`testCount`  | Integer            | The total number of tests to run.
```

Comment:
Please use more strict types that match the standard.

`Date`
`Array of String`
`Number`

You can then use the description to specify that this number is an integer. Although, there seems to be no need for this in this case - we are talking about something users get rather than what they are expected to provide.